addons/purchase/static/src/js/purchase_portal_sidebar.js

```diff
@@ -0,0 +1,112 @@
+odoo.define('purchase.PurchasePortalSidebar', function (require) {
```

Comment:
As this file is a duplicate of `sale_portal_sidebar.js`, it could be better to put the common share directly in the `portal` module (as it's a dependence of both `sale` and `purchase`) to avoid duplicate code.
It could even be put in the `portal_sidebar.js` file directly but need to check if `account` want this sidebar feature too, or otherwise find a solution to make it optional.

Reply:
It could be also an opportunity to update this code (more readability, less call to jQuery/underscore.js).